### PR TITLE
Requests: Sceneitem manipulation requests

### LINKED ISSUE #179: Feature request: Create new scene items
##### Issue type
Feature request

##### Description
My request is to add the feature to create new scene items (image; browsersource; text) and add them to a scene.

##### Technical information
- **Operating System** : Windows 8.1
- **OBS Studio version** : 20.1.3

### PR DESCRIPTION
**Pull request by tt2468**
Similar to #341 but starts from scratch, implementing all of the checks that need to be done.

Added requests:
- `AddSceneItem`
- `GetSceneItemList`

Closes #179

### CODE REVIEW

**Comment by adamschachne** [comment]
Unless I am understanding incorrectly, AddSceneItem is only capable of duplicating a source. Is the intention to be able to create a new source?

**Comment by tt2468** [comment]
@adamschachne that functionality will be handled by another request type. Probably `AddSource` or something. Remember that sceneitems are not sources.

File: src/WSRequestHandler_SceneItems.cpp [hunk 1]
```diff
@@ -544,6 +623,59 @@ RpcResponse WSRequestHandler::DeleteSceneItem(const RpcRequest& request) {
...
+	if (request.hasField("setVisible")) {
+		data.setVisible = obs_data_get_bool(request.parameters(), "setVisible");
+	}
+
+	obs_enter_graphics();
+	obs_scene_atomic_update(scene, AddSourceHelper, &data);
+	obs_leave_graphics();
+
+	OBSDataAutoRelease responseData = obs_data_create();
+	obs_data_set_int(responseData, "itemID", obs_sceneitem_get_id(data.sceneItem));
```
Comment (Palakis) [inline src/WSRequestHandler_SceneItems.cpp:674]:
```suggestion
	obs_data_set_int(responseData, "itemId", obs_sceneitem_get_id(data.sceneItem));
```

File: src/WSRequestHandler_SceneItems.cpp [hunk 2]
```diff
@@ -2,6 +2,85 @@
...
+/**
+* Get a list of all scene items in a scene.
+*
+* @param {String} `sceneName` Name of the scene to get the list of scene items from.
+*
+* @return {Array<Object>} `sceneItems` Array of scene items
+* @return {int} `sceneItems.*.itemId` Unique item id of the source item
+* @return {String} `sceneItems.*.sourceId` ID if the scene item's source. For example `vlc_source` or `image_source`
+* @return {String} `sceneItems.*.sourceName` Name of the scene item's source
+* @return {String} `sceneItems.*.sourceType` Type of the scene item's source. Either `input` or `scene`
```
Comment (Palakis) [inline src/WSRequestHandler_SceneItems.cpp:26]: Scene items can either be inputs, scenes or groups.

File: src/WSRequestHandler_SceneItems.cpp [hunk 3]
```diff
@@ -2,6 +2,85 @@
...
+*
```
Comment (Palakis) [inline src/WSRequestHandler_SceneItems.cpp:27]: Missing: a `sourceKind` attribute on scene item info. (e.g.: browser_source, text_gdiplus)

File: src/WSRequestHandler_SceneItems.cpp (hunk 2 above)
Comment (Palakis) [reply src/WSRequestHandler_SceneItems.cpp:26]: If you implement `sourceKind`, specifying `groups` will probably not be necessary

**Pull request merged**